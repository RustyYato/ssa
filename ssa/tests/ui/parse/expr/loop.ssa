let foo = 
    loop {
        let x = x;
    }
;
