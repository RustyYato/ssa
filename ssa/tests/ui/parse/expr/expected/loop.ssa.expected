[
  Item {
    id: ItemId(1)
    kind: ItemKind::Let(Let {
        binding: Ident {
          id: IdentId(1)
          name: "foo"
        }
        ty: 
        value: Option::Some(Expr {
            id: ExprId(2)
            kind: ExprKind::Loop(Loop {
                block: Block {
                  stmts: [
                    Stmt {
                      id: StmtId(1)
                      kind: StmtKind::Let(Let {
                          binding: Ident {
                            id: IdentId(2)
                            name: "x"
                          }
                          ty: 
                          value: Option::Some(Expr {
                              id: ExprId(1)
                              kind: ExprKind::Ident(Ident {
                                  id: IdentId(3)
                                  name: "x"
                                }
                              )
                            }
                          )
                        }
                      )
                    }
                  ]
                  expr: 
                }
              }
            )
          }
        )
      }
    )
  }
]
